src/content/drafts/custom-formatter-blog-post.md

```diff
@@ -0,0 +1,351 @@
+---
+layout: post
+title: Creating a Custom Formatter 
+teaser: | 
+  Learn how to create a Custom Formatter to customize the ESLint output
+  in this post.
+authors:
+  - bpmutter
+categories:
+  - Tutorials
+tags:
+  - Guest Post
+  - Custom Formatter
+---
+
+In this post, we'll look at how to create [custom formatter](https://eslint.org/docs/latest/developer-guide/working-with-custom-formatter)
+for your ESLint project. Custom formatters are JavaScript functions
+that let you create a unique output for your linting results. 
+
+You might want to make a custom formatter if the formatters built into ESLint 
+don't meet the needs of your use case. You can learn more about ESLint's built-in
+formatters in the [formatter documentation](https://eslint.org/docs/latest/user-guide/formatters/). Some reasons to create a custom formatter include: 
+
+* You only want to report specific error types.
+* You want to format to the results in a way not supported by a built-in formatter.
+* You want to perform an async operation to the results, like sending them to a
+  server for further analysis.
+
+By the end of this post, you'll know how to: 
+
+* Create an ESLint custom formatter
+* Publish it to npm
+* Use it in your project
+
+## What You Will Build
+
+In the remainder of this post, we're going to create a custom formatter that
+outputs ESLint results to [TOML](https://toml.io/).
+TOML is a format for representing data, similar to JSON or YAML.
+Developers often use TOML for config files, as it's optimized for human readability.
+
+The custom formatter we'll build will take an ESLint formatter `results`object
+like this: 
+
+```js
+{
+    "extends": "eslint:recommended",
+    "rules": {
+        "consistent-return": 2,
+        "indent"           : [1, 4],
+        "no-else-return"   : 1,
+        "semi"             : [1, "always"],
+        "space-unary-ops"  : 2
+    }
+}
+```
+
+Then, the custom formatter will create the TOML output: 
+
+```toml
+extends = "eslint:recommended"
+
+[rules]
+consistent-return = 2
+indent = [ 1, 4 ]
+no-else-return = 1
+semi = [ 1, "always" ]
+space-unary-ops = 2
+```
+
+We'll also see how to publish the formatter to npm and use it in a project.
+
+## Steps
+
+Before you begin:
+
+1. Have Node.js and npm installed in your development environment.
+2. Understanding the basics of Node.js.
+3. If you want to publish your custom formatter to npm like we'll cover below, 
+   [create an npm account](https://www.npmjs.com/signup) and login from the
+   [npm CLI](https://docs.npmjs.com/cli/v7/commands/npm-adduser).
+
+### 1. Create Project
+
+First, we're going to create the project for our custom formatter.
+Since an ESLint custom formatter is just a JavaScript function, let's create a 
+new project for the custom formatter by running the following:
+
+```bash
+mkdir eslint-formatter-toml
+cd eslint-formatter-toml
+npm init -y
+```
+
+Add ESLint to the project:
+
+```bash
+npm install eslint --save-dev
```

Comment:
We typically encourage people to run this command instead. 

```suggestion
npm init @eslint/config
```

This will also set up a basic configuration file, so you can remove the next paragraph.

Reply:
ok. also added a bit about selecting the relevant CLI options.